app/views/locations/_form.html.erb

```diff
@@ -190,6 +190,7 @@
     <% end %>
     
     <% if params[:reload] %>setTimeout(function() { location.reload(); }, 3000);<% end %>
+    notyNotification('relax', 3000, 'success', 'topCenter', 'Location Saved');
```

Comment:
Hi, this look great! However, shouldn't we try to connect this to a `success` result from `addTag()`? 

That function is defined here: 

https://github.com/publiclab/plots2/blob/876d0fc084064aaecc23f8003630d7d1ab858fa1/app/assets/javascripts/tagging.js#L4

I think we might have to add a `callback` parameter so that we can send in a function to be executed upon successful completion... it would then have to be passed into this function as well:

https://github.com/publiclab/plots2/blob/876d0fc084064aaecc23f8003630d7d1ab858fa1/app/assets/javascripts/submit_form_ajax.js#L4

I'm happy to help with this, what do you think?

Reply:
Hey @jywarren that's a great idea, shows the difference between novice and veteran developer lol. I would very much appreciate your help, however I would like to give it a try first, If I get stuck , I will surely ask for your help. Will that be fine?

Reply:
@gauravahlawat81 I'd be happy to help too!

Reply:
Thanks, very much appreciated, but i am good. :smile:

Reply:
Hey @jywarren , what if we simply add `notyNotification('relax', 3000, 'success', 'topCenter', 'Location Saved');` to the line number 15, how much different it would be from adding a `callback` param to the methods ?
https://github.com/publiclab/plots2/blob/876d0fc084064aaecc23f8003630d7d1ab858fa1/app/assets/javascripts/submit_form_ajax.js#L4-L15

Reply:
Also I am sorry, it looks like the PR is messed up. I was trying to rebase my code, and looks like I messed something up. Once I solve the issue, I guess I will create a new PR :disappointed:

Reply:
thanks, i think a possible issue with adding the notification within `sendFormSubmissionAjax` is that that function is generalized and so the message may not always be `Location Saved` but could be distinct for different types of form responses. Using a callback function can help whatever upstream code is invoking `sendFormSubmissionAjax()` to decide what specific message ought to be shown!

Reply:
Oh yeah, makes sense.But I am having a hard time creating a callback, do we have some existing examples in our code so that maybe I can look there for help ? 

Thanks for explaining it to me, super helpful !

Reply:
So, we'd start by modifying `addTag()` here:
```js
 function addTag(tagname, submitTo, responseEl = "", callback) { 
```

then when it calls `sendFormSubmissionAjax()` we'd pass the callback in; we'd have to modify that too:

```js
 function sendFormSubmissionAjax(dataObj, submitTo, responseEl = "", callback) { 
```

it'll get passed in, and if it's nil, thats ok, we'll detect it later. Finally, in the `success` listener here within `sendFormSubmissionAjax()`, we'll detect if there's a callback and we'll run it:

```js
     success: (event, success) => { 
       if (callback) callback(event);
```

we might as well pass in the 'event' object, but it's not strictly necessary. Then whatever we send to `addTag(tagname, submitTo, responseEl, callback)` will run `callback()` upon completing successfully. Does that make sense? If not, it's OK - callbacks are complex and a bit old-fashioned, to be honest... all this nesting and passing is one reason there are better styles of programming these days. But if you like I can add some commits here to help you figure it out!

Reply:
Hi @gauravahlawat81 👋🏾 
I have created another [PR ](https://github.com/publiclab/plots2/pull/9113) to continue the work you began here which has been helpful in learning what the issue was... please review it when you get the chance...